### PR TITLE
[GSoC] Added symbolic Expectation, Variance and CrossCovariance matrix

### PR DESCRIPTION
**Pull request by Smit-create**
Added symbolic Expectation, Variance and Covariance matrix
#### References to other Issues or PRs

References from #19299 

#### Other comments
- [x] Add classses
- [x] Add docs and tests 

#### Release Notes

* stats
  * Added Expectation Matrix, Variance Matrix and CrossCovariance Matrix

ping @czgdp1807  @Upabjojr

### CODE REVIEW

**Comment by Upabjojr** [comment]
There are various name possibilities to choose, from wikipedia:

1. https://en.wikipedia.org/wiki/Covariance_matrix

> covariance matrix (also known as auto-covariance matrix, dispersion matrix, variance matrix, or variance–covariance matrix)

2. OTOH, the cross-covariance matrix: https://en.wikipedia.org/wiki/Cross-covariance_matrix

In principle, 1 and 2 are the generalizations of variance and covariance to vector random variables. There are problems with the terminology though, as the _covariance matrix_ is the generalization of the _variance_ (not the _covariance_), while the _cross-covariance matrix_ is the generalization of the _covariance_.

**Comment by czgdp1807** [comment]
Tests?

**Comment by Smit-create** [comment]
I am writing them currently, and push by tonight.

**Comment by Smit-create** [comment]
I have added docs and tests for VarianceMatrix and CrossCovarianceMatrix. I am currently trying to resolve the following error with ExpectationMatrix.
```python
>>> A = MatrixSymbol("A", k, k)
>>> X = RandomMatrixSymbol("X", k, 1)
>>> Expectation((Mul.fromiter([A*X]))) # correct
ExpectationMatrix(A*X)
>>> Expectation((Mul.fromiter([X])))
ExpectationMatrix(X)
>>> A*Expectation((Mul.fromiter([X]))) # this looks incorrect
A*X  # Expected as: A*ExpectationMatrix(X)
```

**Comment by czgdp1807** [comment]
The problem is with `doit` of `ExpectationMatrix`,
```python
>>> r = Expectation((Mul.fromiter([X])))
>>> r
ExpectationMatrix(X)
>>> r.doit()
X
```

**Comment by Smit-create** [comment]
> The problem is with `doit` of `ExpectationMatrix`,

Opps, got it, MatMul will call `doit`, so for `ExpecationMatrix`, `Expectation`'s `doit` is called, so need to overwrite it to return `self `until we implement `doit` for `ExpectationMatrix`. Thanks!!

**Comment by czgdp1807** [comment]
LGTM. Any thoughts @Upabjojr @jmig5776 ?

**Comment by Smit-create** [comment]
ping @Upabjojr

File: sympy/stats/tests/test_symbolic_multivariate.py [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+    ])
+
+    assert Expectation(Expectation(x1 + x2)).doit(deep=False) == ExpectationMatrix(x1 + x2)
+    assert Expectation(Expectation(x1 + x2)).doit() == Matrix([[12, 14], [24, 26]])
+
+
+def test_multivariate_variance():
+    raises(ShapeError, lambda: Variance(A))
+
+    expr = Variance(a)  # type: VarianceMatrix
```
Comment (Upabjojr) [inline sympy/stats/tests/test_symbolic_multivariate.py:87]: maybe also an `assert isinstance(expr, VarianceMatrix)`?

**Comment by Upabjojr** [comment]
This looks good to me.

**Pull request merged**